flux.csl

```diff
@@ -0,0 +1,454 @@
+        <text macro="issued-year"/>
+        <text macro="citation-locator"/>
+      </group>
+    </layout>
+  </citation>
+  <bibliography et-al-use-last="true" entry-spacing="0" line-spacing="2" hanging-indent="true">
+    <sort>
+      <key macro="author"/>
+      <key macro="issued-sort" sort="ascending"/>
+      <key macro="title"/>
+    </sort>
+    <layout>
+      <choose>
+        <if type="article-journal article-magazine article-newspaper" match="any">
+          <group delimiter=", ">
+            <text macro="author"/>
+            <text macro="issued"/>
+            <text macro="title"/>
+            <text macro="container-title"/>
+            <choose>
+              <if type="article-newspaper">
+                <text macro="day-month"/>
+              </if>
+            </choose>
+            <text macro="locators"/>
+          </group>
+        </if>
+        <else-if type="book report" match="any">
+          <group delimiter=", ">
+            <text macro="author"/>
+            <text macro="issued"/>
+            <text macro="title"/>
+            <choose>
+              <if type="book">
+                <text macro="publisher"/>
+              </if>
+            </choose>
+          </group>
+        </else-if>
+        <else-if type="book" match="any">
+          <group delimiter=", ">
+            <text macro="author"/>
+            <text macro="issued"/>
+            <text macro="title"/>
+            <text macro="publisher"/>
+          </group>
+        </else-if>
+        <else-if type="chapter" match="any">
+          <group delimiter=", ">
+            <text macro="author"/>
+            <text macro="issued"/>
+            <text macro="title"/>
+            <group delimiter=" ">
+              <text term="in" font-style="italic" text-case="lowercase" suffix=" :"/>
+              <group delimiter=", ">
+                <text macro="container-contributors"/>
+                <text macro="container-title"/>
+              </group>
+            </group>
+            <text macro="publisher"/>
+            <text macro="locators"/>
+          </group>
+        </else-if>
+        <else-if type="paper-conference" match="any">
+          <group delimiter=", ">
+            <text macro="author"/>
+            <text macro="issued-year"/>
+            <text macro="title"/>
+            <text macro="event"/>
+          </group>
+        </else-if>
+        <else-if type="report" match="any">
+          <group delimiter=", ">
+            <text macro="author"/>
+            <text macro="issued"/>
+            <text macro="title"/>
+          </group>
+        </else-if>
+        <else-if type="thesis">
+          <group delimiter=", " suffix=".">
+            <text macro="author"/>
+            <text macro="issued"/>
+            <text macro="title"/>
+            <text macro="thesis-details"/>
+          </group>
+        </else-if>
+        <else>
+          <group delimiter=", ">
+            <text macro="author"/>
```

Comment:
You're calling the same 3 macros, author, issued, title for each item type. Take these out of the conditionals. Unnecessary repetion.